File: backend/WMSApi/Models/PurchaseOrderItem/PurchaseOrderItem.cs
namespace WMSApi.Models;

public class PurchaseOrderItem
{
    public long Id { get; set; }
    public long ItemId { get; set; }
    public required Item Item { get; set; }
    public long PurchaseOrderId { get; set; }
    public required PurchaseOrder PurchaseOrder { get; set; }
    public long? PalletId { get; set; }
    public Pallet? Pallet { get; set; }
    public required double PurchasedQuantity { get; set; }
    public double CurrentQuantity { get; set; }
    public double Weight { get; set; }
    public required double UnitPrice { get; set; }
    public double MarkupPrice { get; set; }
    public double MarginPrice { get; set; }
    public double FreightPrice { get; set; }
    public double SellPrice { get; set; }
}

File: frontend/wms-client/src/models/purchase-order-item.ts
import Item from "./item";
import PurchaseOrder from "./purchase-order";

interface PurchaseOrderItem {
    id?: number,
    itemId: number,
    item?: Item,
    purchaseOrderId?: number,
    purchaseOrder?: PurchaseOrder,
    palletId?: number,
    pallet?: any,
    purchasedQuantity: number,
    currentQuantity?: number,
    weight?: number,
    unitPrice: number,
    markupPrice: number,
    marginPrice?: number,
    freightPrice?: number,
    sellPrice?: number
}

export default PurchaseOrderItem;